pkg/eestream/rs_test.go

```diff
@@ -545,3 +545,85 @@ func readAllStalled(readers []io.Reader, stalled int) ([][]byte, error) {
 	}
 	return pieces, nil
 }
+
+func BenchmarkReedSolomonErasureScheme(b *testing.B) {
+	data := randData(8 << 20)
+	output := make([]byte, 8<<20)
+
+	confs := []struct{ required, total int }{
+		{2, 4},
+		{30, 60},
+		{50, 80},
+	}
+
+	tests := []struct {
+		name     string
+		dataSize int
+	}{
+		{"1KB", 1 << 10},
+		{"256KB", 256 << 10},
+		{"1MB", 1 << 20},
+		{"4MB", 5 << 20},
+		{"8MB", 8 << 20},
+	}
+
+	for _, conf := range confs {
+		confname := fmt.Sprintf("r%dt%d/", conf.required, conf.total)
+		for _, test := range tests {
+			dataSize := (test.dataSize / conf.required) * conf.required
+			forwardErrorCode, _ := infectious.NewFEC(conf.required, conf.total)
+			erasureScheme := NewRSScheme(forwardErrorCode, 8*1024)
```

Comment:
How about parameterizing the erasure share size too? My experience so far is that we have a huge performance degradation with tiny size like 16 bytes.

Reply:
This test doesn't seem to be affected by that parameter.